src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeService.java

```diff
@@ -80,10 +85,45 @@ public void setStatus(Integer id, DisputeStatus disputeStatus) {
 	 */
 	public void setStatus(Integer id, DisputeStatus disputeStatus, String rejectedReason) {
 		Dispute dispute = disputeRepository.findById(id).orElseThrow();
-		dispute.setStatus(disputeStatus);
-		if (DisputeStatus.REJECTED.equals(disputeStatus)) {
-			dispute.setRejectedReason(rejectedReason);
+
+		if (disputeStatus == null) {
```

Comment:
we should check this before fetching the dispute. Avoid doing work if the call is invalid.

Reply:
Good call.
Done.